airbyte-integrations/connectors/source-enquire-labs/source_enquire_labs/source.py

```diff
@@ -0,0 +1,209 @@
+stream from a source. This pattern is the same one used by Airbyte internally to implement connectors.
+
+The approach here is not authoritative, and devs are free to use their own judgement.
+
+There are additional required TODOs in the files within the integration_tests folder and the spec.json file.
+"""
+
+
+# Basic full refresh stream
+class EnquireLabsStream(HttpStream, ABC):
+
+    primary_key = None
+    url_base = "https://app.enquirelabs.com/api/"
+
+    def __init__(self, secret_key, **kwargs):
+        super().__init__(**kwargs)
+        self.secret_key = secret_key
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        TODO: Override this method to define a pagination strategy. If you will not be using pagination, no action is required - just return None.
+
+        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
+        to most other methods in this class to help you form headers, request bodies, query params, etc..
+
+        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
+        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
+        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
+
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        """
+        TODO: Override this method to define any query parameters to be set. Remove this method if you don't need to define request params.
+        Usually contains common params e.g. pagination size etc.
+        """
+
+        return {}
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {"Authorization": self.secret_key}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        TODO: Override this method to define how a response is parsed.
+        :return an iterable containing each record in the response
+        """
+        return [response.json()]
+
+    def path(
+        self,
+        *,
```

Comment:
What is the purpose of this arg?

Reply:
this adds by default in params when we define this method in `HTTP-API-SOURCE`